src/qw/md.py

```diff
@@ -28,3 +30,401 @@ def text_under_heading(text: str, heading: str) -> str:
         raise QwError(msg)
 
     return "\n".join(sub_heading_lines).strip()
+
+
+# Breaking Markdown text into paragraphs of some type.
+# Picking out fenced code:
+FENCED_RE = re.compile(
+    r"""
+    (?:^|\n+)        # The start of a line
+    (?:~{3,}|`{3,})  # At least three backticks or tildes for the first fence
+    [ \t]*(.*?)\n    # Capturing the format name after the first fence
+    ([\S\s]*?)\n     # Capturing the fenced code: [\S\s] = Anything (including newlines)
+    (?:~{3,}|`{3,})  # At least three backticks or tildes for the second fence
+    (?:$|\n+)        # The end of a line
+    """,
+    re.VERBOSE,
+)
+# Paragraph delimiters. We don't test for numbers at the start of
+# lines because numbered lists must be separated from other paragraphs
+# by a blank line anyway.
+PARAGRAPH_RE = re.compile(
+    r"""
+    \n{2,}|  # a blank line
+    \n(?=\s*[\*\+\-]\s)  # a bullet at the start of the line.
+    """,
+    re.VERBOSE,
+)
+
+# Determinig the type of a paragraph
+## - item
+LISTU_RE = re.compile(r"(\s*)[\*\+\-]\s+(.*)")
+## 1. item
+LISTO_RE = re.compile(r"(\s*)\d+\.\s+(.*)")
+
+# Breaking Markdown paragraphs into runs of some type
+## **bold**
+ASTERISK2_RE = re.compile(
+    r"""
+    \*\*(\S(?:.*\S)?)\*\*  # ** surrounding (captured) nonspaces surrounding text
+    (?=[^*])               # not followed by *, so ***text*** -> bold *text*
```

Comment:
Oh I didn't know that this worked and would ignore the comments, that's cool

Reply:
Only if you use re.VERBOSE